ui/packages/consul-ui/app/components/consul/node/agentless-notice/index.js

```diff
@@ -0,0 +1,34 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+const DISMISSED_VALUE = 'true';
+
+export default class AgentlessNotice extends Component {
+  storageKey = 'consul-nodes-agentless-notice-dismissed';
+  @tracked hasDismissedNotice = false;
+
+  constructor(owner, args) {
+    super(owner, args);
+
+    if (this.args.postfix) {
+      this.storageKey = `consul-nodes-agentless-notice-dismissed-${this.args.postfix}`;
+    }
+
+    if (window.localStorage.getItem(this.storageKey) === DISMISSED_VALUE) {
```

Comment:
Three things that came to my mind when looking at this:

1. Accessing local-storage directly and not wrapping it will make the test-suite really persist values when we dismiss a notice
2. We are checking against a string, although we are setting a boolean when dismissing
3. We aren't really deriving state, but checking local storage and then setting state based on that, which kind of goes against the reactive state ideas of Octane.

What I could imagine doing is creating a wrapper service for local-storage that works with storage classes that encapsulate the buckets of storage we want to interact with. This sounds complicated and may be overkill for a use-case like this, but I suppose we have other places in the application that could benefit from an approach like this.

I prototyped a mechanism that is heavily inspired by [ember-local-storage](https://github.com/funkensturm/ember-local-storage), but works with tracked properties in a repository of mine:

https://github.com/LevelbossMike/local-storage-service

I also demonstrate how this mechanism could be used in testing so that we don't pollute local-storage when executing the test-suite.

Having said that, I'm unsure, if it would be worth the effort to add a mechanism like this for this particular feature. How do you feel about this?

Reply:
@LevelbossMike  I originally was trying to use [ember-tracked-local-storage](https://leadfeeder.github.io/ember-tracked-local-storage/#/how-to-use-it) to get the reactive nature of it, instead of reading that localstorage state and setting local state based on it.  I ran into issues with setting up dynamic keys with the tracked property. However, that doesn't look like it would be an issue in your package.

Ember question for you. I don't see where the `notices` get tracked in your [example](https://github.com/LevelbossMike/local-storage-service/commit/505ca07e78f3f8a26314e6e9bb678963aeac5e13). Do changes to services state trigger changes in the same way that arguments or tracked properties do?

Reply:
Just leaving this here for history. As discussed briefly in a call: In the example repository, I'm setting up the `notices`-storage so that it uses [a tracked array](https://github.com/LevelbossMike/local-storage-service/blob/main/app/storages/notices.js#L11) internally. That's the reason we can create reactive getters based on the state-property of the notices-storage.

Reference: https://github.com/tracked-tools/tracked-built-ins